rust/tools/qltest.sh

```diff
@@ -1,10 +1,22 @@
 #!/bin/bash
 
 set -eu
+set -o pipefail
 
 export RUST_BACKTRACE=full
 QLTEST_LOG="$CODEQL_EXTRACTOR_RUST_LOG_DIR"/qltest.log
-if ! "$CODEQL_EXTRACTOR_RUST_ROOT/tools/$CODEQL_PLATFORM/extractor" --qltest >> "$QLTEST_LOG" 2>&1; then
-  cat "$QLTEST_LOG"
+mkdir -p "$CODEQL_EXTRACTOR_RUST_SCRATCH_DIR"
+TMP_OUT="$(mktemp --tmpdir="$CODEQL_EXTRACTOR_RUST_SCRATCH_DIR" qltest-XXXXXX.log))"
```

Comment:
I wouldn't bother with the `mktemp` and `trap` commands. The scratch dir is cleaned up already when a database is finalized.